src/Plugins/Reminder.php

```diff
@@ -0,0 +1,357 @@
+            ){ return yield $this->unset($command); }
+
+            return yield $this->setReminder($command);
+        });
+    }
+
+    public function apologize(ChatRoom $room, $timedOutReminders): \Generator
+    {
+        if(count($timedOutReminders) > 0){
+            foreach ($timedOutReminders as $key){
+                $value = yield $this->storage->get($key,$room);
+                $text = $value['text'];
+                $name = $value['username'];
+
+                if (null !== $pingableName = yield $this->chatClient->getPingableName($room, $name)) {
+                    $target = "@{$pingableName}";
+                    $reply = $target . " I guess I'm late: " . $text;
+                    yield $this->storage->unset($key, $room);
+                    return $this->chatClient->postMessage($room, $reply, PostFlags::ALLOW_PINGS);
+                }
+            }
+        }
+    }
+
+    public function reschedule(ChatRoom $room, array $upcomingReminders): \Generator
+    {
+        if(count($upcomingReminders) > 0){
+            $this->watchers = [];
+
+            foreach ($upcomingReminders as $key){
+                $value = yield $this->storage->get($key,$room);
+                $text  = $value['text'];
+                $name  = $value['username'];
+                $stamp = $value['timestamp'];
+                $seconds = $stamp - time();
+
+                if (null !== $pingableName = yield $this->chatClient->getPingableName($room, $name)) {
+                    $target = "@{$pingableName}";
+                    $reply = ($seconds <= 0) ? $target ." I guess I'm late: " . $text : $target . " " . $text;
+
+                    if ($seconds <= 0): return $this->chatClient->postMessage($room, $reply); endif;
+
+                    $this->watchers[] = once(function () use ($room, $key, $value, $reply) {
+                        yield $this->storage->unset($key, $room);
+                        return $this->chatClient->postMessage($room, $reply);
+                    }, $seconds * 1000);
+                }
+            }
+        }
+    }
+
+    public function enableForRoom(ChatRoom $room, bool $persist = true){
+        $timedOutReminders = $upcomingReminders = [];
+        $reminders = yield $this->storage->getAll($room);
+
+        if(count($reminders) > 0){
+            foreach ($reminders as $key => $value){
+                $timestamp = $value['timestamp'];
+                $seconds = $timestamp - time();
+
+                if ($seconds <= 0){
+                    $timedOutReminders[] = base_convert($value['id'], 10, 36);
+                } else {
+                    $upcomingReminders[] = base_convert($value['id'], 10, 36);
+                }
+            }
+        }
+        yield from $this->reschedule($room, $upcomingReminders);
+        yield from $this->apologize($room, $timedOutReminders);
+    }
+
+    public function disableForRoom(ChatRoom $room, bool $persist = false){
+        $watchers = $this->watchers;
```

Comment:
No need for this local variable. Just use `$this->watchers` in the following code in this method.